Source/WebGPU/WebGPU/Queue.mm

```diff
@@ -486,6 +486,38 @@ static bool validateWriteTexture(const WGPUImageCopyTexture& destination, const
     m_device.instance().scheduleWork(WTFMove(workItem));
 }
 
+id<MTLFence> Queue::generateFenceToWaitFor()
+{
+    id<MTLFence> fence = [m_device.device() newFence];
+    if (!m_pendingFences)
+        m_pendingFences = [NSMutableArray array];
+
+    [m_pendingFences addObject:fence];
+    return fence;
+}
+
+static void waitForAndClearPendingFences(NSMutableArray<id<MTLFence>> *pendingFences, auto callback)
+{
+    for (id<MTLFence> fence in pendingFences)
+        callback(fence);
+
+    [pendingFences removeAllObjects];
```

Comment:
This assumes that fences added to one command encoder are all used in the command encoder processed next. Could there ever be a situation where WGPU would need to wait for different fences in a compute and render encoder from the same command encoder? If so, I'd like to see a `FIXME` at least.

Reply:
Since I moved away from fences, I will not worry about this situation for now